File: qt/gen_qwineventnotifier.cpp
// +build ignore

#include <QChildEvent>
#include <QEvent>
#include <QMetaMethod>
#include <QMetaObject>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QTimerEvent>
#include <QWinEventNotifier>
#include <qwineventnotifier.h>
#include "gen_qwineventnotifier.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

class MiqtVirtualQWinEventNotifier : public virtual QWinEventNotifier {
public:

	MiqtVirtualQWinEventNotifier(): QWinEventNotifier() {};
	MiqtVirtualQWinEventNotifier(HANDLE hEvent): QWinEventNotifier(hEvent) {};
	MiqtVirtualQWinEventNotifier(QObject* parent): QWinEventNotifier(parent) {};
	MiqtVirtualQWinEventNotifier(HANDLE hEvent, QObject* parent): QWinEventNotifier(hEvent, parent) {};

	virtual ~MiqtVirtualQWinEventNotifier() = default;

	// cgo.Handle value for overwritten implementation
	intptr_t handle__Event = 0;

	// Subclass to allow providing a Go implementation
	virtual bool event(QEvent* e) override {
		if (handle__Event == 0) {
			return QWinEventNotifier::event(e);
		}
		
		QEvent* sigval1 = e;

		bool callback_return_value = miqt_exec_callback_QWinEventNotifier_Event(this, handle__Event, sigval1);

		return callback_return_value;
	}

	// Wrapper to allow calling protected method
	bool virtualbase_Event(QEvent* e) {

		return QWinEventNotifier::event(e);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__EventFilter = 0;

	// Subclass to allow providing a Go implementation
	virtual bool eventFilter(QObject* watched, QEvent* event) override {
		if (handle__EventFilter == 0) {
			return QWinEventNotifier::eventFilter(watched, event);
		}
		
		QObject* sigval1 = watched;
		QEvent* sigval2 = event;

		bool callback_return_value = miqt_exec_callback_QWinEventNotifier_EventFilter(this, handle__EventFilter, sigval1, sigval2);

		return callback_return_value;
	}

	// Wrapper to allow calling protected method
	bool virtualbase_EventFilter(QObject* watched, QEvent* event) {

		return QWinEventNotifier::eventFilter(watched, event);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__TimerEvent = 0;

	// Subclass to allow providing a Go implementation
	virtual void timerEvent(QTimerEvent* event) override {
		if (handle__TimerEvent == 0) {
			QWinEventNotifier::timerEvent(event);
			return;
		}
		
		QTimerEvent* sigval1 = event;

		miqt_exec_callback_QWinEventNotifier_TimerEvent(this, handle__TimerEvent, sigval1);

		
	}

	// Wrapper to allow calling protected method
	void virtualbase_TimerEvent(QTimerEvent* event) {

		QWinEventNotifier::timerEvent(event);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__ChildEvent = 0;

	// Subclass to allow providing a Go implementation
	virtual void childEvent(QChildEvent* event) override {
		if (handle__ChildEvent == 0) {
			QWinEventNotifier::childEvent(event);
			return;
		}
		
		QChildEvent* sigval1 = event;

		miqt_exec_callback_QWinEventNotifier_ChildEvent(this, handle__ChildEvent, sigval1);

		
	}

	// Wrapper to allow calling protected method
	void virtualbase_ChildEvent(QChildEvent* event) {

		QWinEventNotifier::childEvent(event);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__CustomEvent = 0;

	// Subclass to allow providing a Go implementation
	virtual void customEvent(QEvent* event) override {
		if (handle__CustomEvent == 0) {
			QWinEventNotifier::customEvent(event);
			return;
		}
		
		QEvent* sigval1 = event;

		miqt_exec_callback_QWinEventNotifier_CustomEvent(this, handle__CustomEvent, sigval1);

		
	}

	// Wrapper to allow calling protected method
	void virtualbase_CustomEvent(QEvent* event) {

		QWinEventNotifier::customEvent(event);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__ConnectNotify = 0;

	// Subclass to allow providing a Go implementation
	virtual void connectNotify(const QMetaMethod& signal) override {
		if (handle__ConnectNotify == 0) {
			QWinEventNotifier::connectNotify(signal);
			return;
		}
		
		const QMetaMethod& signal_ret = signal;
		// Cast returned reference into pointer
		QMetaMethod* sigval1 = const_cast<QMetaMethod*>(&signal_ret);

		miqt_exec_callback_QWinEventNotifier_ConnectNotify(this, handle__ConnectNotify, sigval1);

		
	}

	// Wrapper to allow calling protected method
	void virtualbase_ConnectNotify(QMetaMethod* signal) {

		QWinEventNotifier::connectNotify(*signal);

	}

	// cgo.Handle value for overwritten implementation
	intptr_t handle__DisconnectNotify = 0;

	// Subclass to allow providing a Go implementation
	virtual void disconnectNotify(const QMetaMethod& signal) override {
		if (handle__DisconnectNotify == 0) {
			QWinEventNotifier::disconnectNotify(signal);
			return;
		}
		
		const QMetaMethod& signal_ret = signal;
		// Cast returned reference into pointer
		QMetaMethod* sigval1 = const_cast<QMetaMethod*>(&signal_ret);

		miqt_exec_callback_QWinEventNotifier_DisconnectNotify(this, handle__DisconnectNotify, sigval1);

		
	}

	// Wrapper to allow calling protected method
	void virtualbase_DisconnectNotify(QMetaMethod* signal) {

		QWinEventNotifier::disconnectNotify(*signal);

	}

};

QWinEventNotifier* QWinEventNotifier_new() {
	return new MiqtVirtualQWinEventNotifier();
}

QWinEventNotifier* QWinEventNotifier_new2(HANDLE hEvent) {
	return new MiqtVirtualQWinEventNotifier(hEvent);
}

QWinEventNotifier* QWinEventNotifier_new3(QObject* parent) {
	return new MiqtVirtualQWinEventNotifier(parent);
}

QWinEventNotifier* QWinEventNotifier_new4(HANDLE hEvent, QObject* parent) {
	return new MiqtVirtualQWinEventNotifier(hEvent, parent);
}

void QWinEventNotifier_virtbase(QWinEventNotifier* src, QObject** outptr_QObject) {
	*outptr_QObject = static_cast<QObject*>(src);
}

QMetaObject* QWinEventNotifier_MetaObject(const QWinEventNotifier* self) {
	return (QMetaObject*) self->metaObject();
}

void* QWinEventNotifier_Metacast(QWinEventNotifier* self, const char* param1) {
	return self->qt_metacast(param1);
}

struct miqt_string QWinEventNotifier_Tr(const char* s) {
	QString _ret = QWinEventNotifier::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QWinEventNotifier_SetHandle(QWinEventNotifier* self, HANDLE hEvent) {
	self->setHandle(hEvent);
}

HANDLE QWinEventNotifier_Handle(const QWinEventNotifier* self) {
	return self->handle();
}

bool QWinEventNotifier_IsEnabled(const QWinEventNotifier* self) {
	return self->isEnabled();
}

void QWinEventNotifier_SetEnabled(QWinEventNotifier* self, bool enable) {
	self->setEnabled(enable);
}

struct miqt_string QWinEventNotifier_Tr2(const char* s, const char* c) {
	QString _ret = QWinEventNotifier::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

struct miqt_string QWinEventNotifier_Tr3(const char* s, const char* c, int n) {
	QString _ret = QWinEventNotifier::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}

void QWinEventNotifier_override_virtual_Event(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__Event = slot;
}

bool QWinEventNotifier_virtualbase_Event(void* self, QEvent* e) {
	return ( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_Event(e);
}

void QWinEventNotifier_override_virtual_EventFilter(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__EventFilter = slot;
}

bool QWinEventNotifier_virtualbase_EventFilter(void* self, QObject* watched, QEvent* event) {
	return ( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_EventFilter(watched, event);
}

void QWinEventNotifier_override_virtual_TimerEvent(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__TimerEvent = slot;
}

void QWinEventNotifier_virtualbase_TimerEvent(void* self, QTimerEvent* event) {
	( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_TimerEvent(event);
}

void QWinEventNotifier_override_virtual_ChildEvent(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__ChildEvent = slot;
}

void QWinEventNotifier_virtualbase_ChildEvent(void* self, QChildEvent* event) {
	( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_ChildEvent(event);
}

void QWinEventNotifier_override_virtual_CustomEvent(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__CustomEvent = slot;
}

void QWinEventNotifier_virtualbase_CustomEvent(void* self, QEvent* event) {
	( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_CustomEvent(event);
}

void QWinEventNotifier_override_virtual_ConnectNotify(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__ConnectNotify = slot;
}

void QWinEventNotifier_virtualbase_ConnectNotify(void* self, QMetaMethod* signal) {
	( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_ConnectNotify(signal);
}

void QWinEventNotifier_override_virtual_DisconnectNotify(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQWinEventNotifier*>( (QWinEventNotifier*)(self) )->handle__DisconnectNotify = slot;
}

void QWinEventNotifier_virtualbase_DisconnectNotify(void* self, QMetaMethod* signal) {
	( (MiqtVirtualQWinEventNotifier*)(self) )->virtualbase_DisconnectNotify(signal);
}

void QWinEventNotifier_Delete(QWinEventNotifier* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQWinEventNotifier*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qwineventnotifier.go
package qt

import (
	"unsafe"
)

type QWinEventNotifier struct {
	h          uintptr
	isSubclass bool
}

// NewQWinEventNotifier constructs a new QWinEventNotifier object.
func NewQWinEventNotifier() *QWinEventNotifier {

	ret := newQWinEventNotifier(QWinEventNotifier_new())
	ret.isSubclass = true
	return ret
}

// NewQWinEventNotifier2 constructs a new QWinEventNotifier object.
func NewQWinEventNotifier2(hEvent HANDLE) *QWinEventNotifier {

	ret := newQWinEventNotifier(QWinEventNotifier_new2(hEvent))
	ret.isSubclass = true
	return ret
}

// NewQWinEventNotifier3 constructs a new QWinEventNotifier object.
func NewQWinEventNotifier3(parent *QObject) *QWinEventNotifier {

	ret := newQWinEventNotifier(QWinEventNotifier_new3(parent.cPointer()))
	ret.isSubclass = true
	return ret
}

// NewQWinEventNotifier4 constructs a new QWinEventNotifier object.
func NewQWinEventNotifier4(hEvent HANDLE, parent *QObject) *QWinEventNotifier {

	ret := newQWinEventNotifier(QWinEventNotifier_new4(hEvent, parent.cPointer()))
	ret.isSubclass = true
	return ret
}

func (this *QWinEventNotifier) MetaObject() *QMetaObject {
	return newQMetaObject(QWinEventNotifier_MetaObject(this.h))
}

func (this *QWinEventNotifier) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QWinEventNotifier_Metacast(this.h, param1_Cstring))
}

func QWinEventNotifier_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QWinEventNotifier_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QWinEventNotifier) SetHandle(hEvent HANDLE) {
	QWinEventNotifier_SetHandle(this.h, hEvent)
}

func (this *QWinEventNotifier) Handle() HANDLE {
	xxxxxxxxx
}

func (this *QWinEventNotifier) IsEnabled() bool {
	return (bool)(QWinEventNotifier_IsEnabled(this.h))
}

func (this *QWinEventNotifier) SetEnabled(enable bool) {
	QWinEventNotifier_SetEnabled(this.h, (bool)(enable))
}

func QWinEventNotifier_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QWinEventNotifier_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QWinEventNotifier_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QWinEventNotifier_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QWinEventNotifier) callVirtualBase_Event(e *QEvent) bool {

	return (bool)(QWinEventNotifier_virtualbase_Event(unsafe.Pointer(this.h), e.cPointer()))

}
func (this *QWinEventNotifier) OnEvent(slot func(super func(e *QEvent) bool, e *QEvent) bool) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_Event(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_Event
func miqt_exec_callback_QWinEventNotifier_Event(self QWinEventNotifier, cb intptr_t, e *QEvent) bool {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(e *QEvent) bool, e *QEvent) bool)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQEvent(e)

	virtualReturn := gofunc((&QWinEventNotifier{h: self}).callVirtualBase_Event, slotval1)

	return (bool)(virtualReturn)

}

func (this *QWinEventNotifier) callVirtualBase_EventFilter(watched *QObject, event *QEvent) bool {

	return (bool)(QWinEventNotifier_virtualbase_EventFilter(unsafe.Pointer(this.h), watched.cPointer(), event.cPointer()))

}
func (this *QWinEventNotifier) OnEventFilter(slot func(super func(watched *QObject, event *QEvent) bool, watched *QObject, event *QEvent) bool) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_EventFilter(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_EventFilter
func miqt_exec_callback_QWinEventNotifier_EventFilter(self QWinEventNotifier, cb intptr_t, watched *QObject, event *QEvent) bool {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(watched *QObject, event *QEvent) bool, watched *QObject, event *QEvent) bool)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQObject(watched)

	slotval2 := newQEvent(event)

	virtualReturn := gofunc((&QWinEventNotifier{h: self}).callVirtualBase_EventFilter, slotval1, slotval2)

	return (bool)(virtualReturn)

}

func (this *QWinEventNotifier) callVirtualBase_TimerEvent(event *QTimerEvent) {

	QWinEventNotifier_virtualbase_TimerEvent(unsafe.Pointer(this.h), event.cPointer())

}
func (this *QWinEventNotifier) OnTimerEvent(slot func(super func(event *QTimerEvent), event *QTimerEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_TimerEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_TimerEvent
func miqt_exec_callback_QWinEventNotifier_TimerEvent(self QWinEventNotifier, cb intptr_t, event *QTimerEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *QTimerEvent), event *QTimerEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQTimerEvent(event)

	gofunc((&QWinEventNotifier{h: self}).callVirtualBase_TimerEvent, slotval1)

}

func (this *QWinEventNotifier) callVirtualBase_ChildEvent(event *QChildEvent) {

	QWinEventNotifier_virtualbase_ChildEvent(unsafe.Pointer(this.h), event.cPointer())

}
func (this *QWinEventNotifier) OnChildEvent(slot func(super func(event *QChildEvent), event *QChildEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_ChildEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_ChildEvent
func miqt_exec_callback_QWinEventNotifier_ChildEvent(self QWinEventNotifier, cb intptr_t, event *QChildEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *QChildEvent), event *QChildEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQChildEvent(event)

	gofunc((&QWinEventNotifier{h: self}).callVirtualBase_ChildEvent, slotval1)

}

func (this *QWinEventNotifier) callVirtualBase_CustomEvent(event *QEvent) {

	QWinEventNotifier_virtualbase_CustomEvent(unsafe.Pointer(this.h), event.cPointer())

}
func (this *QWinEventNotifier) OnCustomEvent(slot func(super func(event *QEvent), event *QEvent)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_CustomEvent(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_CustomEvent
func miqt_exec_callback_QWinEventNotifier_CustomEvent(self QWinEventNotifier, cb intptr_t, event *QEvent) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(event *QEvent), event *QEvent))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQEvent(event)

	gofunc((&QWinEventNotifier{h: self}).callVirtualBase_CustomEvent, slotval1)

}

func (this *QWinEventNotifier) callVirtualBase_ConnectNotify(signal *QMetaMethod) {

	QWinEventNotifier_virtualbase_ConnectNotify(unsafe.Pointer(this.h), signal.cPointer())

}
func (this *QWinEventNotifier) OnConnectNotify(slot func(super func(signal *QMetaMethod), signal *QMetaMethod)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_ConnectNotify(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_ConnectNotify
func miqt_exec_callback_QWinEventNotifier_ConnectNotify(self QWinEventNotifier, cb intptr_t, signal *QMetaMethod) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(signal *QMetaMethod), signal *QMetaMethod))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQMetaMethod(signal)

	gofunc((&QWinEventNotifier{h: self}).callVirtualBase_ConnectNotify, slotval1)

}

func (this *QWinEventNotifier) callVirtualBase_DisconnectNotify(signal *QMetaMethod) {

	QWinEventNotifier_virtualbase_DisconnectNotify(unsafe.Pointer(this.h), signal.cPointer())

}
func (this *QWinEventNotifier) OnDisconnectNotify(slot func(super func(signal *QMetaMethod), signal *QMetaMethod)) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QWinEventNotifier_override_virtual_DisconnectNotify(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QWinEventNotifier_DisconnectNotify
func miqt_exec_callback_QWinEventNotifier_DisconnectNotify(self QWinEventNotifier, cb intptr_t, signal *QMetaMethod) {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(signal *QMetaMethod), signal *QMetaMethod))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQMetaMethod(signal)

	gofunc((&QWinEventNotifier{h: self}).callVirtualBase_DisconnectNotify, slotval1)

}
